### PR TITLE
langref: add `usingnamespace` example usage for separate impl block

### PR DESCRIPTION
**Pull request by expikr**

### CODE REVIEW

File: doc/langref.html.in [hunk 1]
```diff
@@ -6926,6 +6926,27 @@ pub usingnamespace @cImport({
...
+      </p>
+      {#syntax_block|zig|separate_impl.zig#}
+const Wine = struct {
+    name: [:0]const u8,
+    year: isize,
+    price: usize,
+    origin: Country,
+    pub usingnamespace _impl_Wine;
+};
+const _impl_Wine = opaque {
```
Comment (rootbeer) [inline doc/langref.html.in:6940]: The langref probably should not encourage underscore prefixed names?  This name should probably be CamelCase, too?  (`WineImpl` or `WinePrivate`?)

File: doc/langref.html.in [hunk 2]
```diff
@@ -6926,6 +6926,27 @@ pub usingnamespace @cImport({
...
+    const Self = Wine;
+    pub fn importTax(self: Self, destCountry: Country) {
+        // ...
+    }
+};
+      {#end_syntax_block#}
+      <p>
+      This may be useful when for reusing the same behaviors across different variants of a data structure.
```
Comment (rootbeer) [inline doc/langref.html.in:6948]: Can you briefly add text (or in a separate PR) explaining the type relationships between `Wine` and `_impl_Wine`?  E.g, an instance of `Wine` isn't also an instance of `_impl_Wine`?  (Is it more like the fields of `_impl_Wine` got textually added to `Wine`, right and they're not related types?)

**Review by rootbeer — COMMENTED** [review]
This looks like a helpful addition to the langref to me.  I've got some comments, but they're all minor.

File: doc/langref.html.in [hunk 3]
```diff
@@ -6926,6 +6926,27 @@ pub usingnamespace @cImport({
...
+      <p>
+      Another use case is splitting up the declaration of a data structure separately from its implementation:
+      </p>
+      {#syntax_block|zig|separate_impl.zig#}
+const Wine = struct {
+    name: [:0]const u8,
+    year: isize,
+    price: usize,
+    origin: Country,
+    pub usingnamespace _impl_Wine;
```
[the remark this answers is not in the document]
Comment (rohlem) [reply doc/langref.html.in:6938]: To be exact, the `declarations` are embedded / aliased, the `fields` of `_impl_Wine`, if any, are ignored.
(`langref` currently only uses the word "member" to refer to errors in error sets and fields.)

File: doc/langref.html.in (hunk 2 above)
Comment (rohlem) [reply doc/langref.html.in:6948]: Only declarations are added, not fields. (But makes sense to clarify this.)

**Comment by andrewrk** [comment]
I do not wish to encourage this feature's use by suggesting any more ways to use it.

**Pull request closed without merging**